src/cpp/rtps/RTPSDomain.cpp

```diff
@@ -275,6 +277,19 @@ RTPSWriter* RTPSDomain::createRTPSWriter(
         WriterAttributes& watt,
         WriterHistory* hist,
         WriterListener* listen)
+{
+    return RTPSDomain::createRTPSWriter(p, watt,
+            BasicPayloadPool::get(PoolConfig::from_history_attributes(hist->m_att)),
```

Comment:
This will not work correctly, as BasicPayloadPool is designed assuming the `get_payload` method will be called when the `CacheChangePool` is created.

I would create a private `template <Functor> create_writer` method so that we could specialize the call to the corresponding method on the `RTPSParticipantImpl`

Reply:
done